pymc3_hmm/step_methods.py

```diff
@@ -109,6 +110,27 @@ def ffbs_astep(gamma_0: np.ndarray, Gammas: np.ndarray, log_lik: np.ndarray):
     return samples
 
 
+def traverse_graph_and_replace(graph):
+    visited = []
+
+    queue = [graph]
+    count = 0
+    while queue:
+        node = queue.pop(0)
+        if node not in visited:
+            visited.append(node)
+            if node.__str__() =="Alloc.0":
+                shared_var = shared(node.eval(), name = f'shared_var_{count}', borrow = True)
+                clone(graph, {node:shared_var})
+                count += 1
+            if node.owner is not None:
+                inputs = node.owner.inputs
+                for input in inputs:
+                    queue.append(input)
+    return graph
```

Comment:
We can condition this functionality on the presence of a `SwitchingProcess` log-likelihood.  That would prevent us from needing to search the graph, and it would tell us exactly where the `Alloc` variable is, then we could simply use `clone` to replace it (and make whatever other in-place changes we require).